Client/mods/deathmatch/logic/CClientGif.h

```diff
@@ -0,0 +1,129 @@
+/*****************************************************************************
+ *
+ *  PROJECT:     Multi Theft Auto
+ *  LICENSE:     See LICENSE in the top level directory
+ *
+ *  Multi Theft Auto is available from https://www.multitheftauto.com/
+ *
+ *****************************************************************************/
+
+#pragma once
+
+#include "CClientGifDisplay.h"
+
+enum {
+    CGif_None = 0,
+    CGif_Current = 1,
+    CGif_Background = 2,
+    CGif_Previous = 3
+};
+
+class CClientGifRect {
+public:
+    uint left = 0;
+    uint top = 0;
+    uint width = 0;
+    uint height = 0;
+    bool valid = false;
+    CClientGifRect() = default;
+    CClientGifRect(const uint& l, const uint& t, const uint& w, const uint& h) : left(l), top(t), width(w), height(h) { valid = true; }
+};
+
+class CClientGifLoader {
+public:
+    CClientGifLoader() = default;
+    CClientGifLoader(SString&);
+    CClientGifLoader(uint8_t*,unsigned long);
+    CClientGifLoader(char*);
+    void        CreateFrame(std::vector<unsigned char*>&,std::vector<int>&);
+    void        Load(std::vector<unsigned char*>&,std::vector<int>&,long);
+    uint8_t*    GetBuffer() { return buffer; }
+    long&       GetWidth() { return width; }
+    long&       GetHeight() { return height; }
+    long&       GetDelay() { return delay; }
+    long&       GetFrameNumber() { return frameNumber; }
+    long&       GetFrameCount() { return frameCount; }
+    SString     GetFormat() { return format; }
+    bool        operator!();
+
+    static long SkipChunk(uint8_t**, long);
+    static long LoadHeader(unsigned, uint8_t**, void**, unsigned, long*, long);
+    static long LoadFrame(uint8_t**, long*, uint8_t*, uint8_t*);
+
+private:
+    struct {
+        uint8_t r;
+        uint8_t g;
+        uint8_t b;
+    }* pallete;
+    SString        format = "Undefined";
+    uint8_t*       buffer = nullptr;
+    uint8_t*       address = nullptr;
+    void*          palleteDecoder = nullptr;
+    void*          palleteDecoderPrevious = nullptr;
+    long           size = 0;
```

Comment:
Also all class members must have prefix m_ to explitly identify them.

Reply:
because those were private i defined them like that?
is changing them important?

Reply:
This is the code style: https://github.com/multitheftauto/mtasa-blue/wiki/Style-Guide

Reply:
fixed!